core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -122,13 +122,23 @@ public static void write(TableMetadata metadata, OutputFile outputFile) {
   public static void internalWrite(
       TableMetadata metadata, OutputFile outputFile, boolean overwrite) {
     boolean isGzip = Codec.fromFileName(outputFile.location()) == Codec.GZIP;
-    OutputStream stream = overwrite ? outputFile.createOrOverwrite() : outputFile.create();
-    try (OutputStream ou = isGzip ? new GZIPOutputStream(stream) : stream;
-        OutputStreamWriter writer = new OutputStreamWriter(ou, StandardCharsets.UTF_8)) {
-      JsonGenerator generator = JsonUtil.factory().createGenerator(writer);
-      generator.useDefaultPrettyPrinter();
-      toJson(metadata, generator);
-      generator.flush();
+    try (OutputStream os = overwrite ? outputFile.createOrOverwrite() : outputFile.create()) {
+      if (isGzip) {
+        try (OutputStream gos = new GZIPOutputStream(os);
+            OutputStreamWriter osw = new OutputStreamWriter(gos, StandardCharsets.UTF_8)) {
+          JsonGenerator generator = JsonUtil.factory().createGenerator(os);
+          generator.useDefaultPrettyPrinter();
+          toJson(metadata, generator);
+          generator.flush();
+        }
+      } else {
+        try (OutputStreamWriter osw = new OutputStreamWriter(os, StandardCharsets.UTF_8)) {
```

Comment:
I am not convinced we need the two code branches for this, nor that it's desired.
`OutputStream::close` must obey general `Closeable::close` contract which requries the implementation to be idempotent.

Thus, if we had

```
OutputStream stream = overwrite ? outputFile.createOrOverwrite() : outputFile.create();
    try (OutputStream ou = isGzip ? new GZIPOutputStream(stream) : stream;
```

we now should be able to:

```
try (OutputStream stream = overwrite ? outputFile.createOrOverwrite() : outputFile.create();
   OutputStream ou = isGzip ? new GZIPOutputStream(stream) : stream;
```

guaranteeing all streams to be closed, regardless at which point an exception is thrown.

Can you try that?

Reply:
thanks!

> Is there a gradle target to auto-format?

```
./gradlew -DallModules=true spotlessApply
```

( see also https://gist.github.com/findepi/da2ea85d095eaa0f3cfdcbb311ff59ed#reformat-code )